tests/core/numeric-utils/test_integer_squareroot.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+
+from eth.utils.numeric import (
+    integer_squareroot,
+)
+
+
+@pytest.mark.parametrize(
+    'value,expected',
+    (
+        (0, 0),
+        (1, 1),
+        (2, 1),
+        (3, 1),
+        (4, 2),
+        (27, 5),
+        (65535, 255),
+        (65536, 256),
+        (18446744073709551615, 4294967295),
+        (1.5, ValueError()),
+        (-1, ValueError()),
+    )
+)
+def test_integer_squareroot(value, expected):
```

Comment:
Noting that the `integer_squareroot` function is a good candidate for property based testing since it has a very clearly defined property between input/output `output ** 2 <= input`.